Fix dynamic properties and other problems in tests for PHP 8.2

come-nc opened the pull request:
See #32595 
See #32463 

Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

come-nc committed:
Fix dynamic properties and other problems in tests for PHP 8.2

Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

come-nc committed:
Fix errors from PHP 8.2 testing

Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

blizzz commented on tests/lib/Encryption/Keys/StorageTest.php:
```diff
@@ -31,22 +31,23 @@
 	protected $storage;
 
-	/** @var \PHPUnit\Framework\MockObject\MockObject */
+	/** @var MockObject */
 	protected $util;
```
probably this wants to become a union type? And the following ones.

blizzz reviewed (commented):
a few remarks, also CI isn't convinced.

tcitworld commented on apps/dav/lib/RootCollection.php:
```diff
@@ -118,7 +118,6 @@ public function __construct() {
 		$roomCalendarRoot->disableListing = $disableListing;
 
 		$publicCalendarRoot = new PublicCalendarRoot($caldavBackend, $l10n, $config, $logger);
-		$publicCalendarRoot->disableListing = $disableListing;
```
Why this one specifically and not the ones above?

come-nc replied:
The other classes have a `disableListing` property, but `PublicCalendarRoot` does not. It’s because it does not inherit the same Sabre class I think.

blizzz commented on apps/dav/tests/unit/BackgroundJob/PruneOutdatedSyncTokensJobTest.php:
```diff
@@ -34,13 +34,14 @@
 
 class PruneOutdatedSyncTokensJobTest extends TestCase {
+	/** @var ITimeFactory | MockObject */
+	private $timeFactory;
```
never mind

come-nc commented on apps/dav/tests/unit/BackgroundJob/CleanupInvitationTokenJobTest.php:
```diff
@@ -78,7 +76,7 @@ public function testRun() {
 			]);
 
-		$function = $this->createMock(IQueryFunction::class);
+		$function = 'function1337';
```
It needs to be a string, I think it’s supposed to be SQL, but as the tests were using a mock I considered it did not need to be valid SQL. I’ll check the tests.

blizzz replied:
A follow up comment did not make it, is was to suggest to also call it `functionToken` as you do later, for i was irritated first and thought this was some debugging left over or something. And only with the other occurrence i paid more attention and figured what you say here.

come-nc committed:
Fix syntax in VerificationTokenTest.php

Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

come-nc committed:
Add Union types for properties in tests/lib/Encryption/Keys/StorageTest.php

Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

come-nc committed:
Fix tests depending on a trailing whitespace

Many editors remove trailing whitespaces, so avoid depending on them for
 tests success.

Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

szaimen reviewed (approved):
🐘
